library/core/src/iter/adapters/fuse.rs

```diff
@@ -198,8 +198,27 @@ impl<I: Default> Default for Fuse<I> {
     /// let iter: Fuse<slice::Iter<'_, u8>> = Default::default();
     /// assert_eq!(iter.len(), 0);
     /// ```
+    ///
+    /// Note that if `I`'s default value is not an empty iterator, then this will not be
+    /// an empty iterator.
```

Comment:
IMO this is pointing out an arbitrary case that is in no way surprising (at least not to someone unfamiliar with the implementation history). I think what would be more helpful is to say that this function is equivalent to `IteratorType::default().fused()`.

Reply:
Not mentioning non-empty `I::default()` at all would IMO make the doctest seem like a non-sequitur (and the doctest should remain so this change doesn't get accidentally undone later), so I think it should still be mentioned as an aside. 

I changed this to say 

> This is equivalent to `I::default().fuse()`; e.g. if `I::default()` is not an empty iterator, then this will not be an empty iterator.

with a footnote that the equivalence assumes `I::fuse` is not overridden. Does that sound good?